x/tally/keeper/integration_helpers_test.go

```diff
@@ -0,0 +1,287 @@
+
+	proof = f.generateCommitProof(t, stakerKey, drID, commitment, res.Height)
+	_, err = f.contractKeeper.Execute(
+		f.Context(),
+		f.coreContractAddr,
+		staker,
+		[]byte(fmt.Sprintf(commitMsg, drID, commitment, stakerPubKey, proof)),
+		sdk.NewCoins(sdk.NewCoin(bondDenom, math.NewIntFromUint64(1))),
+	)
+	require.NoError(t, err)
+
+	proof = f.generateRevealProof(t, stakerKey, drID, commitment, res.Height)
+	_, err = f.contractKeeper.Execute(
+		f.Context(),
+		f.coreContractAddr,
+		staker,
+		[]byte(fmt.Sprintf(revealMsg, drID, drID, stakerPubKey, proof)),
+		sdk.NewCoins(sdk.NewCoin(bondDenom, math.NewIntFromUint64(1))),
+	)
+	require.NoError(t, err)
+
+	return res.DrID
+}
+
+var addToAllowListMsg = `{
+	"add_to_allowlist": {
+	  "public_key": "%s"
+    }
+}`
+
+var stakeMsg = `{
+  "stake": {
+    "public_key": "%s",
+    "proof": "%s",
+    "memo": "YWRkcmVzcw=="
+  }
+}`
+
+var postDataRequestMsg = `{
+	"post_data_request": {
+	  "posted_dr": {
+		"version": "0.0.1",
+		"exec_program_id": "%s",
+		"exec_inputs": "ZXhlY19pbnB1dHM=",
+		"exec_gas_limit": 10,
+		"tally_program_id": "%s",
+		"tally_inputs": "dGFsbHlfaW5wdXRz",
+		"tally_gas_limit": 10,
+		"replication_factor": 1,
+		"consensus_filter": "AA==",
+		"gas_price": "10",
+		"memo": "%s"
+	  },
+	  "seda_payload": "",
+	  "payback_address": "AQID"
+	}
+}`
+
+var commitMsg = `{
+  "commit_data_result": {
+    "dr_id": "%s",
+    "commitment": "%s",
+    "public_key": "%s",
+    "proof": "%s"
+  }
+}`
+
+var revealMsg = `{
+  "reveal_data_result": {
+    "dr_id": "%s",
+    "reveal_body": {
+      "id": "%s",
+      "salt": "9c0257114eb9399a2985f8e75dad7600c5d89fe3824ffa99ec1c3eb8bf3b0501",
+      "exit_code": 0,
+      "gas_used": 0,
+      "reveal": "Ghkvq84TmIuEmU1ClubNxBjVXi8df5QhiNQEC5T8V6w=",
+      "proxy_public_keys": []
+    },
+    "public_key": "%s",
+    "proof": "%s",
+    "stderr": [],
+    "stdout": []
+  }
+}`
+
+// generateStakeProof generates a proof for a stake message given a
+// base64-encoded memo.
+func (f *fixture) generateStakeProof(t *testing.T, signKey []byte) string {
+	// TODO
+	// var sequence uint64 = 0
+
+	memo := "YWRkcmVzcw=="
+	memoBytes, err := base64.StdEncoding.DecodeString(memo)
+	require.NoError(t, err)
+
+	// Create slices for each component
+	stakeBytes := []byte("stake")
+
+	hasher := sha3.NewLegacyKeccak256()
+	hasher.Write(memoBytes)
+	memoHash := hasher.Sum(nil)
+
+	chainIDBytes := []byte(f.chainID)
+	contractAddrBytes := []byte(f.coreContractAddr.String())
+
+	sequenceBytes := make([]byte, 16)
+	// binary.BigEndian.PutUint64(sequenceBytes, sequence) // TODO
+
+	allBytes := append([]byte{}, stakeBytes...)
+	allBytes = append(allBytes, memoHash...)
+	allBytes = append(allBytes, chainIDBytes...)
+	allBytes = append(allBytes, contractAddrBytes...)
+	allBytes = append(allBytes, sequenceBytes...)
+
+	hasher.Reset()
+	hasher.Write(allBytes)
+	hash := hasher.Sum(nil)
+
+	proof, err := vrf.NewK256VRF().Prove(signKey, hash)
```

Comment:
If not mistaken, currently we are using a Hash of the Reveal Body.

Reply:
For commitments, right? I saw that the contract additionally requires proofs for stake, commit, and reveal txs.

Reply:
yep yep!! 👍